geolocation-API/getCurrentPosition_permission_allow.https.html

```diff
@@ -15,12 +15,16 @@
       return test_driver.set_permission({ name: "geolocation" }, "prompt");
     });
     await test_driver.set_permission({ name: "geolocation" }, "granted");
-    const promise = new Promise((resolve, reject) => {
-      navigator.geolocation.getCurrentPosition(resolve, reject);
+    let hasMethodReturned = false;
+    const position = await new Promise((resolve, reject) => {
+      navigator.geolocation.getCurrentPosition((position) => {
```

Comment:
I think this callback needs to be wrapped in t.step_func() in order for assert_true() to work properly in it.

Reply:
Suggestion: How about reject()ing instead of the assert? I’m trying to avoid t.step_func() as it adds indirection and makes tests harder (for me😅) to read/understand.

Reply:
I think this is unavoidable when we are trying to make such a precise assertion about execution order. I prefer assert because it makes it clearer what assertions are being made by the test code.

Reply:
Got it, and yeah I agree. Ok, brb!